frontend/src/pages/TaskMapPage.tsx

```diff
@@ -0,0 +1,191 @@
+  taskIds: number[];
+  selectedTask: Task | undefined;
+  setSelectedTask: any;
+}> = ({ taskIds, selectedTask, setSelectedTask }) => {
+  return (
+    <div className={classes(css.taskContainer)}>
+      {taskIds.map((taskId) => {
+        return (
+          <div key={taskId}>
+            <SingleTask
+              taskId={taskId}
+              selected={selectedTask?.id === taskId}
+              setSelectedTask={setSelectedTask}
+            />
+          </div>
+        );
+      })}
+    </div>
+  );
+};
+
+const CustomNodeComponent: FC<{ data: any }> = ({ data }) => {
+  return <div>{data.label}</div>;
+};
+
+const nodeTypes = {
+  special: CustomNodeComponent,
+};
+
+export const TaskMapPage = () => {
+  const tasks = useSelector(allTasksSelector(), shallowEqual);
+  const dispatch = useDispatch<StoreDispatchType>();
+  const taskRelations = groupTaskRelations(tasks);
+  const currentRoadmap = useSelector<RootState, Roadmap | undefined>(
+    chosenRoadmapSelector,
+    shallowEqual,
+  );
+  const [selectedTask, setSelectedTask] = useState<Task | undefined>(undefined);
+
+  const groups = taskRelations.map(({ synergies }, idx) => ({
+    id: `${idx}`,
+    type: 'special',
+    sourcePosition: Position.Right,
+    targetPosition: Position.Left,
+    draggable: false,
+    position: { x: 550 * idx, y: 80 },
+    data: {
+      label: (
+        <TaskComponent
+          taskIds={synergies}
+          selectedTask={selectedTask}
+          setSelectedTask={setSelectedTask}
+        />
+      ),
+    },
+  }));
+
+  const edges = taskRelations.flatMap(({ dependencies }, idx) =>
+    dependencies.map(({ from, to }) => {
+      const targetGroupIdx = taskRelations.findIndex(({ synergies }) =>
+        synergies.includes(to),
+      );
+
+      return {
+        id: `from-${from}-to-${to}`,
+        source: String(idx),
+        sourceHandle: `from-${from}`,
+        target: String(targetGroupIdx),
+        targetHandle: `to-${to}`,
+        style: { stroke: 'green' },
+      };
+    }),
+  );
+
+  const elements = [...groups, ...edges];
+
+  const onConnect = async (data: any) => {
+    const { sourceHandle, targetHandle } = data;
+    const roadmapId = currentRoadmap?.id;
+    const type = 0; // Type 0 = dependency
+
+    // Handles are in form 'from-{id}' and 'to-{id}', splitting required
+    const from = sourceHandle.includes('from')
+      ? Number(sourceHandle.split('-')[1])
+      : Number(targetHandle.split('-')[1]);
+
+    const to = targetHandle.includes('to')
+      ? Number(targetHandle.split('-')[1])
+      : Number(sourceHandle.split('-')[1]);
+
+    if (!roadmapId) return;
+    await dispatch(
+      roadmapsActions.addTaskRelation({
+        from,
+        to,
+        type,
+        roadmapId,
+      }),
+    );
+    dispatch(roadmapsActions.getRoadmaps());
+  };
+
+  return (
+    <>
+      <ReactFlow
+        className={classes(css.flowContainer)}
+        elements={elements}
+        nodeTypes={nodeTypes}
```

Comment:
Personally, I would add the elements and nodeTypes definitions here as they are not used anywhere else
```suggestion
        elements={[...groups, ...edges]}
        nodeTypes={{
          special: CustomNodeComponent,
        }}
```